go/runtime/bundle/discovery.go

```diff
@@ -29,8 +29,9 @@ const (
 	// requestTimeout is the time limit for http client requests.
 	requestTimeout = 10 * time.Second
 
-	// maxBundleSizeBytes is the maximum allowed bundle size in bytes.
-	maxBundleSizeBytes = 20 * 1024 * 1024 // 20 MB
+	// maxDefaultBundleSizeBytes is the maximum allowed default bundle size
+	// in bytes.
+	maxDefaultBundleSizeBytes = 20 * 1024 * 1024 // 20 MB
 )
```

Comment:
Minor: Will `requestTimeout` of 10s be sufficient for big bundles or do we also want to make this configurable/function of size?

Reply:
I would say for the request to complete since we use it [to set](https://github.com/oasisprotocol/oasis-core/blob/979c1c210b824dbb9c1bb4abac7d46dabd6cc5e5/go/runtime/bundle/discovery.go#L62) `http.Client.Timeout`
```
       // Timeout specifies a time limit for requests made by this
	// Client. The timeout includes connection time, any
	// redirects, and reading the response body. The timer remains
	// running after Get, Head, Post, or Do return and will
	// interrupt reading of the Response.Body.
```

Reply:
What is the max size we anticipate? For the reference I see that our documentation recommends "Recommended: 1 Gbps internet connection with low latency".

Might be a good idea to future proof us by making this parametric...

Reply:
The more I think about it there is also the following risk:

Compute nodes may also run runtimes developed by other entities. I they are intentionally or non-intentionally compromised, a bad actor could cheaply post new deployments versions to registry (L1 transaction) that would reference big bundles. In theory you could then fill-up the compute nodes (that are subscribed/run compromised runtime) disk fast.

Worst case if against suggestion they also run validator instance as part of their compute node you would halt down validator instance + runtimes, otherwise only runtimes.

Now we could limit/monitor the total storage used inside `/bundles` dir to avoid this case (can be also config parameter). Unless we do it per-runtime, there is still a risk of this attack that halt all the runtimes that compute node is running? Finally, in this context it might be useful to have a flag that allows/dis-allows automatic downloads of runtime bundles. 

Unlikely scenario but might be worth protecting us against in the follow-up? @peternose

Reply:
I increased timeout to 1 minute.

Since only runtime bundles that include RONL component and maybe few system components are downloaded through the client, the bundles should not be large, at least not in the near future. The size of the latest sapphire bundle is 7.45 MB, so limits 20 MB and 1 minute should suffice. 

I'm not worried about your threat model because we have very little runtimes in production. If someone can post new deployments to the registry this is a much bigger problem than big bundles. Actually, it would be great if big bundles would stop your runtime until this is resolved. And lastly, bundle cleanup and bundle limitations will ensure that a node will have only the latest and sometimes second-latest runtime bundle stored locally, so this should be approx. 100 MB (including exploded files).